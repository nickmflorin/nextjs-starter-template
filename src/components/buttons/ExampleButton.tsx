import { CSSProperties, ReactNode } from "react";

import classNames from "classnames";

export type ExampleButtonProps = {
  readonly className?: string;
  readonly style?: CSSProperties;
  readonly children: ReactNode;
};

export const ExampleButton = (props: ExampleButtonProps): JSX.Element => (
  <button {...props} className={classNames("example-button", props.className)} />
);
