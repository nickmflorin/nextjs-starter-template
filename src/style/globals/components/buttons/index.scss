@use "partials";

.example-button {
  background-color: partials.$button-bg-color;
}
