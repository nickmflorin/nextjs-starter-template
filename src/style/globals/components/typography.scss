%typography-base {
  font-feature-settings: "tnum" on, "lnum" on;
  font-variant: tabular-nums;
}

html,
body {
  @extend %typography-base;

  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell,
    "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
}
