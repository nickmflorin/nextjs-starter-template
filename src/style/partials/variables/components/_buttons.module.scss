@use "../colors.module" as colors;

$button-bg-color: colors.$gray;
