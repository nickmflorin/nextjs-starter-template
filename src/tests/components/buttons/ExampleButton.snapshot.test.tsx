import React from "react";

import renderer from "react-test-renderer";

import { ExampleButton } from "components/buttons";

describe("example button properly renders", () => {
  it("matches snapshot", () => {
    const tree = renderer.create(<ExampleButton>Test</ExampleButton>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
