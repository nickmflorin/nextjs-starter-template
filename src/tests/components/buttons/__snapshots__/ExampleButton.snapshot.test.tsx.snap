// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example button properly renders matches snapshot 1`] = `
<button
  className="example-button"
>
  Test
</button>
`;
